lib/Connector/AlphaVantageConnector.php

```diff
@@ -118,6 +121,83 @@ public function onDataRequest(WidgetDataRequestEvent $event)
         }
     }
 
+    public function onWidgetEditOption(WidgetEditOptionRequestEvent $event)
+    {
+        $this->getLogger()->debug('onWidgetEditOption');
+
+        // Pull the widget we're working with.
+        $widget = $event->getWidget();
+        if ($widget === null) {
+            throw new NotFoundException();
+        }
+
+        // We handle the twitter widget and the property with id="type"
+        if ($widget->type === 'stocks' && $event->getPropertyId() === 'searchSymbol') {
+
+            if (empty($this->getSetting('apiKey'))) {
+                $this->getLogger()->debug('onWidgetEditOption: AlphaVantage API not configured.');
+                return;
+            }
+
+            try {
+                $results = [];
+                $bestMatches = $this->getSearchResults($event->getPropertyValue());
+
+                if (count($bestMatches) > 0) {
+                    foreach($bestMatches as $match) {
+                        $results[] = [
+                            'name' => implode(' ', [$match['1. symbol'], $match['2. name']]),
+                            'type' => $match['1. symbol'],
+                            'id' => $event->getPropertyId(),
+                        ];
+                    }
+                }
+
+                $event->setOptions($results);
+            } catch (\Exception $exception) {
+                $this->getLogger()->error('onWidgetEditOption: Failed to get symbol search results. e = ' . $exception->getMessage());
+            }
+        }
+    }
+
+    /**
+     * Get Stocks data through symbol search
+     *
+     * @param string $keywords
+     * @return array
+     * @throws GeneralException
+     */
+    private function getSearchResults(string $keywords): array
+    {
+        try {
+            $this->getLogger()->debug('AlphaVantage Connector : getSearchResults is served from the API.');
+
+            $request = $this->getClient()->request('GET', 'https://www.alphavantage.co/query', [
+                'query' => [
+                    'function' => 'SYMBOL_SEARCH',
+                    'keywords' => urlencode($keywords),
+                    'apikey' => $this->getSetting('apiKey')
+                ]
+            ]);
+
+            $data = json_decode($request->getBody(), true);
+
+            if (array_key_exists('bestMatches', $data)) {
+                return $data['bestMatches'];
+            }
+
+            return [];
+
+        } catch (GuzzleException $guzzleException) {
+            throw new GeneralException(
+                'Guzzle exception getting Stocks data . E = ' .
```

Comment:
It is our convention to have the `.` on the next line (which is different to the JS convention)